core/src/main/java/io/questdb/cairo/TableWriterAPI.java

```diff
@@ -132,5 +132,15 @@ void addColumn(
      */
     boolean supportsMultipleWriters();
 
-    void truncate();
+    /**
+     * Truncates table. For non-WAL tables, this method has to be called when the
+     * {@link CairoEngine#lockReaders(TableToken)} lock is held, i.e. when there are no readers reading from the table.
+     *
+     * @param keepSymbolTables defines whether the operation does not or does truncate symbol tables,
+     *                         i.e. internal symbol string to int symbol code mappings. Sometimes the symbols should
+     *                         be kept to make sure that DETACH/ATTACH PARTITION does not lose data for symbol columns.
+     *                         WAL tables always retain the symbol tables.
+     * @throws UnsupportedOperationException when keepSymbolTables is set to false on a WAL table.
+     */
+    void truncate(boolean keepSymbolTables);
```

Comment:
From API readability point of view, I'd have a separate method to truncate keeping symbols. 
Boolean parameters are better to avoid if possible.